source/android/mobile/src/main/java/io/adaptivecards/adaptivecardssample/MainActivityAdaptiveCardsSample.java

```diff
@@ -398,49 +437,36 @@ private String loadFile(Uri uri)
         return null;
     }
 
-    private void loadAdaptiveCard(Intent data)
+    /**
+     * Load given AdaptiveCard
+     * @param fileName name of chosen card JSON file
+     * @param payload file contents
+     */
+    private void loadAdaptiveCard(String fileName, String payload)
     {
-
-        String fullString = loadFile(data.getData());
-        if (fullString.isEmpty())
+        if (payload.isEmpty())
         {
             return;
         }
-        loadAdaptiveCard(fullString);
-
-        EditText fileEditText = (EditText) findViewById(R.id.fileEditText);
-        List path = data.getData().getPathSegments();
-        fileEditText.setText((String)path.get(path.size()-1));
-
-    }
-
-    private void loadAdaptiveCard(String payload)
-    {
+        m_selectedCardText.setText(fileName);
         m_jsonEditText.setText(payload);
 
         // Render it immediately
         renderAdaptiveCard(true);
     }
 
-
-    private void loadHostConfig(Intent data)
+    /**
+     * Load given host config
+     * @param fileName name of chosen host config JSON file
+     * @param hostConfigStr file contents
+     */
+    private void loadHostConfig(String fileName, String hostConfigStr)
```

Comment:
we try not to have data types in variable names (unless it comes from a cast), you could name this as hostConfigJson or something that may be more descriptive